### PR TITLE
Fixes and updates in Kwik model

### LINKED ISSUE #144: Backup the kwik file
When first loading a dataset, backup the kwik file in a `kwik~` file. If the backup file already exists, do nothing.

### LINKED ISSUE #145: Implement kwik model save
* [ ] Save spike clusters
* [ ] Save cluster groups (not colors)

### LINKED ISSUE #162: Implement and test clustering logic in session
* [x] Update the cluster store after merge and split
* [ ] Update the session and views after clustering actions
* [ ] Unit tests
* [ ] Test on a real dataset

### LINKED ISSUE #172: Investigate invalid time error in waveform loader on the big dataset

### LINKED ISSUE #173: Implement and try the wizard on a real dataset

### LINKED ISSUE #179: Take the cluster group into account in the wizard

### LINKED ISSUE #180: Load cluster groups from the kwik file

### LINKED ISSUE #181: spike times should be in seconds
and be converted to int64 when computing CCGs

### LINKED ISSUE #184: The waveform loader should take multiple recordings into account

### LINKED ISSUE #185: Only open the kwik file when loading and saving
The kwik file handle should not remain open, because there is a risk of corruption. The kwx and kwd files can/should be kept open, because:

1. We need dynamic access to the data (especially for the kwd file)
2. They're opened in read-only mode during manual clustering, so no risk of corruption in principle

### LINKED ISSUE #188: Investigate mask bug in waveform view
using dataset (3)

### PR DESCRIPTION
**Pull request by rossant**
Ready for review.

* [x] Renamed `spike_times` to `spike_samples` (in number of samples), and added `spike_times` attribute (in seconds).
* [x] Support multiple recordings in Kwik.
* [x] The kwik file is no longer open at all times. It is re-opened in append mode when saving or changing the shank or clustering.
* [x] Save to kwik (spike clusters and cluster groups)
* [x] Backup the kwik file
* [x] Take `channel_order` into account in MEA, Kwik model, and cluster store. This fixes #188.
* [x] Clustering in session
* [x] Wizard

## Closes

* Closes #144
* Closes #145
* Closes #162
* Closes #172 
* Closes #173
* Closes #179
* Closes #180 
* Closes #181
* Closes #184
* Closes #185
* Closes #188

### CODE REVIEW

**Comment by rossant** [comment]
100 commits: the right time to decree this PR as ready for review/merge. @nippoo

File: phy/cluster/manual/cluster_metadata.py [hunk 1]
```diff
@@ -6,13 +6,11 @@
...
 
-from collections import defaultdict, OrderedDict, MutableMapping
+from collections import defaultdict
 from copy import deepcopy
 
-from ...utils._color import _random_color
-from ...utils._misc import _as_dict, _fun_arg_count, _as_list, _is_list
-from ...ext.six import iterkeys, itervalues, iteritems
-from ._utils import _unique, _spikes_in_clusters, UpdateInfo
+from ...utils._misc import _as_list, _is_list
```
Comment (cgestes) [inline phy/cluster/manual/cluster_metadata.py]: each time you wrote more than one . you kill a kitten and make your code less maintainable :P

Comment (cgestes) [inline phy/cluster/manual/cluster_metadata.py]: http://eatwatchrun.files.wordpress.com/2011/03/cute-kittenweee_thumb.jpg

Comment (nippoo) [inline phy/cluster/manual/cluster_metadata.py]: lol. more than one what?

Comment (cgestes) [inline phy/cluster/manual/cluster_metadata.py]: . dot in a row in an import line :)

**Pull request merged**